src/components/DialogAction/index.jsx

```diff
@@ -44,6 +44,12 @@ function DialogAction(props) {
     destructive,
     ...rest
   } = props;
+  let cancelled = false;
+
+  // Set this so we can avoid calling setActionExecuting if the caller
+  // decides to do something that unmounts us in onComplete.
+  useEffect(() => () => (cancelled = true));
```

Comment:
Could you check if it still works if this is rewritten as `useEffect(() => () => (cancelled = true), []);` to not have the empty useEffect triggered on each render.

Reply:
It looks like this doesn't work. With this form of it, I'm guessing the cleanup handler only exists during the first render, and then disappears.

Reply:
Hm, weird :( https://stackoverflow.com/a/55020668/6150432 suggests that it should run. Does it print a console log statement if you put one in the cleanup function?

Reply:
I'm going to go ahead and merge this as is, but if there's something we should look at as a follow-up to improve this, I'm happy to do so.